api/v1alpha1/adminnetworkpolicy_types.go

```diff
@@ -0,0 +1,290 @@
+
+const (
+	// RuleActionPass enables admins to provide exceptions to ClusterNetworkPolicies and delegate this rule to
+	// K8s NetworkPolicies.
+	AdminNetPolRuleActionPass AdminNetPolRuleAction = "Pass"
+	// RuleActionDeny enables admins to deny specific traffic.
+	AdminNetPolRuleActionDeny AdminNetPolRuleAction = "Deny"
+	// RuleActionAllow enables admins to specifically allow certain traffic.
+	AdminNetPolRuleActionAllow AdminNetPolRuleAction = "Allow"
+)
+
+// AdminNetworkPolicyPeer defines an in-cluster peer to allow traffic to/from.
+// Exactly one of the selector pointers should be set for a given peer.
+type AdminNetworkPolicyPeer struct {
+	// +optional
+	Namespaces *NamespaceSet `json:"namespaces,omitempty"`
+
+	// +optional
+	NamespacedPods *NamespaceAndPodSet `json:"namespacedpods,omitempty"`
+}
+
+// NamespaceSet defines a flexible way to select Namespaces in a cluster.
+// Exactly one of the selectors should be set.  If a consumer observes none of
+// its fields are set, they should assume an option they are not aware of has
+// been specified and fail closed.
+type NamespaceSet struct {
+	// Self cannot be "false" when it is set.
+	// If Self is "true" then all pods in the subject's namespace are selected.
+	// +optional
+	Self *bool `json:"self,omitempty"`
+
+	// NotSelf cannot be "false" when it is set.
+	// if NotSelf is "true" then all pods not in the subject's Namespace are selected.
+	// +optional
+	NotSelf *bool `json:"notself,omitempty"`
+
+	// NamespaceSelector is a labelSelector used to select Namespaces, This field
+	// follows standard label selector semantics; if present but empty, it selects
+	// all Namespaces.
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+
+	// SameLabels is used to select a set of Namespaces that share the same values
+	// for a set of labels.
+	// To be selected a Namespace must have all of the labels defined in SameLabels,
+	// and they must all have the same value as the subject of this policy.
+	// If Samelabels is Empty then nothing is selected.
+	// +optional
+	SameLabels []string `json:"samelabels,omitempty"`
+
+	// NotSameLabels is used to select a set of Namespaces that do not have a set
+	// of label(s). To be selected a Namespace must have none of the labels defined
+	// in NotSameLabels. If NotSameLabels is empty then nothing is selected.
+	// +optional
+	NotSameLabels []string `json:"notsamelabels,omitempty"`
+}
+
+// PodSet defines a flexible way to select pods in a cluster. Exactly one of the
+// selectors should be set.  If a consumer observes none of its fields are set,
+// they should assume an option they are not aware of has been specified and fail closed.
+type PodSet struct {
+	// PodSelector is a labelSelector used to select Pods, This field is NOT optional,
+	// follows standard label selector semantics and if present but empty, it selects
+	// all Pods.
+	PodSelector *metav1.LabelSelector `json:"podselector"`
```

Comment:
I know this is in the KEP, but wondering now if this should just be called "selector" (since the struct already gives us the context that we're selecting pods). 

Same for `NamespaceSet` - WDYT?

Reply:
Remove `PodSet` Per https://github.com/kubernetes-sigs/network-policy-api/pull/30/files#r830356060